src/trace_processor/util/args_utils.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SRC_TRACE_PROCESSOR_UTIL_ARGS_UTILS_H_
+#define SRC_TRACE_PROCESSOR_UTIL_ARGS_UTILS_H_
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "perfetto/base/status.h"
+#include "src/trace_processor/types/variadic.h"
+
+namespace perfetto::trace_processor {
+class TraceStorage;
+
+class ArgSet;
+
+class ArgNode {
```

Comment:
Nothing about this is "arg" specific. Please factor this out into a proper json writer class similar to what I did was jsonparser to make this reusable in other contexts (of which I believe there are many).

Reply:
The context here is that at some point I'm hoping to add support for `timestamp` type to args, so that SDK can write timestamps as arguments and the UI can interpret them, so I have been planning to keep this specific to args, so it can be extended in the future (the 90% of the complexity here is anyway `AppendArg`, which does all of the necessary parsing to convert a flat list of args into a tree).